ios/FluentUI/Drawer/DrawerPresentationController.swift

```diff
@@ -438,13 +436,27 @@ class DrawerPresentationController: UIPresentationController {
         return margins
     }
 
-    private func frameForSeparator(in bounds: CGRect, withThickness thickness: CGFloat) -> CGRect {
-        var bounds = bounds
-        // Separator should stay fixed even when content view is moving - compensating for move
-        if extraContentSize < 0 && extraContentSizeEffectWhenCollapsing == .move {
-            bounds.origin.y += presentationDirection == .down ? -extraContentSize : extraContentSize
+    private func frameForPresentedViewController(in bounds: CGRect) -> CGRect {
+        var frame = CGRect(x: bounds.origin.x,
+                           y: bounds.origin.y,
+                           width: bounds.size.width,
+                           height: bounds.size.height)
+
+        // Moves the presented view controller (drawer) towards its presenting base in the content view if it's being dragged to dismissal.
+        // In case the drawer is being expanded, the content view grows with the gesture extra content size and the drawer keeps its
+        // original offset relative to the content view.
+        let gestureOffset = extraContentSize < 0 && extraContentSizeEffectWhenCollapsing == .move ? extraContentSize : 0
+
+        if presentationDirection.isVertical {
+            frame.origin.y += presentationDirection == .down ? gestureOffset - shadowOffset : -gestureOffset + shadowOffset
+        } else {
+            frame.origin.x += presentationDirection == .fromLeading ? gestureOffset - shadowOffset : -gestureOffset + shadowOffset
         }
 
+        return frame
+    }
+
+    private func frameForSeparator(in bounds: CGRect, withThickness thickness: CGFloat) -> CGRect {
```

Comment:
seems like a logic change for separator? is it ok?

Reply:
Yes. The reason for this change is because the previous logic had the content view (the one that contains the drawer view) moving down with the drawer inside of it. So, the separator needed to remain fixed in place relative to the content position and this logic was intended to compensate for that.

Now that we move the inner view (the drawer inside the content view) we don't need to compensate the separator anymore, so it remains in a fixed place with the content view.